src/LibObjectFile.Tests/Elf/ElfSimpleTests.cs

```diff
@@ -12,6 +13,20 @@ namespace LibObjectFile.Tests.Elf
 {
     public class ElfSimpleTests : ElfTestBase
     {
+        [Test]
+        public void TryReadThrows()
+        {
+            using var stream = File.OpenRead("TestFiles/cmnlib.b00");
+            Assert.False(ElfObjectFile.TryRead(stream, out var elf, out var diagnostics));
+            Assert.NotNull(elf);
+            Assert.AreEqual(
+                "Unable to read entirely program header [3]. Not enough data (size: 32) read at offset 148 from the stream",
```

Comment:
better check directly diagnostics.Count == 1, and DiagnosticId instead of matching strings.

Reply:
Done